opentelemetry-collector/templates/NOTES.txt

```diff
@@ -0,0 +1,11 @@
+{{- $validApiKey := and (.Values.honeycomb.apiKey) (ne .Values.honeycomb.apiKey "YOUR_API_KEY") -}}
+
+{{- if not $validApiKey }}
+##### ERROR:
```

Comment:
naive question, what happens if you hit this error condition? 

also if this is helpful/obvious to users, should we also require dataset be set?

Reply:
dataset has a default name, and the Readme does explain how to change it, though we may want to make that part more prominent. If you don't specify an API key, the collector will throw unauthorized errors (via Honeycomb exporter).